hashintel/hash
# Entity type editor – Show correct properties in property selector, add edit bar, and minor changes

## 🌟 What is the purpose of this PR?




Next stage in entity type editor – see below. 

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203007126736604/1203135635302814/f) _(internal)_

## 🚫 Blocked by




N/A

## 🔍 What does this change?




**Functionality changes:**

- Populate property type selector with existing property types, hide property types already added
- Ensure the property type you click on is the one inserted and displayed
- Temporarily remove inoperable property menu buttons
- Ensure property menu shows the correct version number
- Edit bar – including operable discard changes button
- Add JSON object type for expected values chips
- Tooltip with full path for property type in the property type menu

Code changes:

- New `useInitTypeSystem` hook – and init the type system on the view entity type page
- `useEntityType` hook now takes on `onCompleted` param, which we use to reset the state of the editor whenever the entity type loads
- We no longer use `router.asPath` to resolve the entity type – this fixes a bug where hash or query params breaks finding the entity type
- The state management of changes / UI mode has been moved to simplify the UI
- All property types are now fetched on load and provided via a context to enable the property selector to display them
- New `dataTypeNames` to show the human readable name of data type chips
- The new property type menu no longer inserts a fake property type – this is in preparation for making it actually work

## 📜 Does this require a change to the docs?



N/A

## ⚠️ Known issues



## 🐾 Next steps



## 🛡 What tests cover this?





## ❓ How to test this?



1.  Create an entity type in the new entity type form
1. Ensure you can select property types from the property type selector and that the edit bar appears
2. Ensure you can remove property types
3. Ensure discard changes on the edit bar works
